src/app/app-components/courses/add-course/add-course.component.ts

```diff
@@ -72,16 +92,13 @@ export class AddCourseComponent implements OnInit {
     this.router.navigate(['/courses']);
   }
 
-  onDateChange(date: Date) {
-    this.date = date;
-  }
-
-  onDurationChange(duration: number) {
-    this.length = duration;
+  private parseDateString(dateString: string): string {
+    const date = new Date(dateString);
+    return date.getDate() + '/' + (date.getMonth() + 1).toString().padStart(2, '0') + '/' + date.getFullYear();
```

Comment:
It's a good idea to create the intermediary variables, like `month`, `day` and `year`, otherwise it becomes difficult to understand what's the format in use.

Reply:
Yes, that makes absolute sense - I'll fix it before submitting.